src/answerfile.cpp

```diff
@@ -0,0 +1,426 @@
+                m_ini.getValue("network_external", "subnet_mask"));
+        } catch (const std::runtime_error& e) {
+            throw std::runtime_error(
+                fmt::format("Section 'network_external' field 'subnet_mask' "
+                            "validation failed - {}",
+                    e.what()));
+        }
+    }
+
+    external.domain_name = m_ini.getValue("network_external", "domain_name");
+
+    if (m_ini.exists("network_external", "gateway")) {
+        try {
+            external.gateway = convertStringToAddress(
+                m_ini.getValue("network_external", "gateway"));
+        } catch (const std::runtime_error& e) {
+            throw std::runtime_error(
+                fmt::format("Section 'network_external' field 'gateway' "
+                            "validation failed - {}",
+                    e.what()));
+        }
+    }
+
+    if (m_ini.exists("network_external", "nameservers")) {
+        std::vector<std::string> nameservers;
+        boost::split(nameservers,
+            m_ini.getValue("network_external", "nameservers"),
+            boost::is_any_of(", "), boost::token_compress_on);
+
+        external.nameservers = nameservers;
+    }
+}
+
+void AnswerFile::loadManagementNetwork()
+{
+    management.con_interface
+        = m_ini.getValue("network_management", "interface", false);
+
+    try {
+        management.con_ip_addr = convertStringToAddress(
+            m_ini.getValue("network_management", "ip_address", false));
+    } catch (const std::runtime_error& e) {
+        throw std::runtime_error(
+            fmt::format("Section 'network_management' field 'ip_address' "
+                        "validation failed - {}",
+                e.what()));
+    }
+
+    management.con_mac_addr
+        = m_ini.getValue("network_management", "mac_address");
+
+    try {
+        management.subnet_mask = convertStringToAddress(
+            m_ini.getValue("network_management", "subnet_mask", false));
+    } catch (const std::runtime_error& e) {
+        throw std::runtime_error(
+            fmt::format("Section 'network_management' field 'subnet_mask' "
+                        "validation failed - {}",
+                e.what()));
+    }
+
+    management.domain_name
+        = m_ini.getValue("network_management", "domain_name", false);
+
+    if (m_ini.exists("network_management", "gateway")) {
+        try {
+            management.gateway = convertStringToAddress(
+                m_ini.getValue("network_management", "gateway"));
+        } catch (const std::runtime_error& e) {
+            throw std::runtime_error(
+                fmt::format("Section 'network_management' field 'gateway' "
+                            "validation failed - {}",
+                    e.what()));
+        }
+    }
+
+    if (m_ini.exists("network_management", "nameservers")) {
+        std::vector<std::string> nameservers;
+        boost::split(nameservers,
```

Comment:
Have you tried using `<ranges>` from C++20?
https://en.cppreference.com/w/cpp/ranges/split_view

Reply:
No... Though I don't see how it would help.